include/deal.II/lac/trilinos_tpetra_sparse_matrix.h

```diff
@@ -0,0 +1,724 @@
+       * with <i>M</i> being this matrix.
+       *
+       * Source and destination must not be the same vector.
+       */
+      void
+      vmult_add(Vector<Number> &dst, const Vector<Number> &src) const;
+
+
+      /**
+       * Adding matrix-vector multiplication. Add <i>M<sup>T</sup>*src</i> to
+       * <i>dst</i> with <i>M</i> being this matrix. This function does the same
+       * as vmult_add() but takes the transposed matrix.
+       *
+       * Source and destination must not be the same vector.
+       */
+      void
+      Tvmult_add(Vector<Number> &dst, const Vector<Number> &src) const;
+      /** @} */
+
+      /**
+       * @name Mixed Stuff
+       */
+      /** @{ */
+      /**
+       * Print the matrix to the given stream, using the format (line,col)
+       * value, i.e. one nonzero entry of the matrix per line. The optional flag
+       * outputs the sparsity pattern in Trilinos style, where the data is
+       * sorted according to the processor number when printed to the stream, as
+       * well as a summary of the matrix like the global size.
+       */
+      void
+      print(std::ostream &out,
+            const bool    print_detailed_trilinos_information = false) const;
+
+      /**
+       * This command does two things:
+       * <ul>
+       * <li> If the matrix was initialized without a sparsity pattern, elements
+       * have been added manually using the set() command. When this process is
+       * completed, a call to compress() reorganizes the internal data
+       * structures (sparsity pattern) so that a fast access to data is possible
+       * in matrix-vector products.
+       * <li> If the matrix structure has already been fixed (either by
+       * initialization with a sparsity pattern or by calling compress() during
+       * the setup phase), this command does the %parallel exchange of data.
+       * This is necessary when we perform assembly on more than one (MPI)
+       * process, because then some non-local row data will accumulate on nodes
+       * that belong to the current's processor element, but are actually held
+       * by another. This command is usually called after all elements have been
+       * traversed.
+       * </ul>
+       *
+       * In both cases, this function compresses the data structures and allows
+       * the resulting matrix to be used in all other operations like matrix-
+       * vector products. This is a collective operation, i.e., it needs to be
+       * run on all processors when used in %parallel.
+       *
+       * See
+       * @ref GlossCompress "Compressing distributed objects"
+       * for more information.
+       *
+       * @note The @p operation can be safely omitted, as that parameter is not
+       * used at all and is only present to ensure compatibility with other
+       * SparseMatrix classes.
+       */
+      void
+      compress([[maybe_unused]] VectorOperation::values operation =
+                 VectorOperation::unknown);
```

Comment:
Oh, yes, the `[[maybe_unused]]` has no effect here... I will remove that.

But in this form, it is already compatible with the other classes. Therefore, what is the point in forcing the user to provide an `operation` that is never used? This gives the impression that the `operation` has some impact, but that is not true.

Reply:
Some classes/underlying libraries need to know what operation is being performed. To be compatible between these classes, the argument needs to be on these functions. You are right that perhaps here it can be defaulted. Do the other classes have a default argument on this function?

Reply:
No, 
the other (sparse matrix) classes have no default argument for `operation`. In the case of the deal.II own SparseMatrix class, the function compress() does nothing and still needs an argument for `operation`.

Reply:
Let's remove the default argument here as well then, so that these functions remain exactly the same between all classes that have them. The argument may not be used, but it doesn't hurt either to have to be explicit about the operation one wants.

Reply:
Okay,
I removed that default argument.